src/inmanta/server/services/compilerservice.py

```diff
@@ -68,11 +68,17 @@ async def compile_done(self, compile: data.Compile) -> None:
 
         1- Notifications are delivered at least once (retry until all listeners have returned or raise an exception other
         than CancelledError)
-        2- Notification are delivered out-of-band (i.e. the next compile can already start, multiple notifications can be in
-        flight at any given time, out-of-order delivery is possible but highly unlikely)
-        3- Notification are cancelled upon shutdown
+        2- Notification are delivered blocking or non-blocking (i.e. the next compile can already start or not)
+        3- Multiple notifications can be in flight at any given time, out-of-order delivery is possible but highly unlikely.
+        4- Notification are cancelled upon shutdown
```

Comment:
I find the terminology confusing here.

What about 
-`Receive notification of all completed compiles` -> `Coroutine to execute when this listener is notified of a completed compile`
- `Notification are delivered blocking or non-blocking (i.e. the next compile can already start or not)` -> `If this listener is blocking, the next compile can only start after this coroutine is done`
- `Notification are cancelled upon shutdown` -> `This coroutine is cancelled upon shutdown`